contracts/staking/contracts/src/libs/LibProxy.sol

```diff
@@ -0,0 +1,109 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
+import "./LibStakingRichErrors.sol";
+
+
+library LibProxy {
+
+    enum RevertRule {
+        REVERT_ON_ERROR,
+        ALWAYS_REVERT,
+        NEVER_REVERT
+    }
+
+    /// @dev Executes a read-only call to the staking contract, via `revertDelegateCall`.
+    ///      By routing through `revertDelegateCall` any state changes are reverted.
+    function proxyCall(
+        address destination,
+        RevertRule revertRule,
+        bytes4 customSelector,
```

Comment:
Is it correct to say that if you specify a non-zero `customSelector` then `ignoreSelector` will always be `true`?

Reply:
No—but is the distinction clear if they’re renamed to `customEgressSelector` and `ignoreIngressSelector`?

Reply:
For reference:
```
                                                             | customEgressSelector | ignoreIngressSelector
StakingProxy fallback                        | no                                    | no
ReadOnlyProxy fallback                    | yes                                   | no
ReadOnlyProxy revertDelegateCall  | no                                    | yes
```